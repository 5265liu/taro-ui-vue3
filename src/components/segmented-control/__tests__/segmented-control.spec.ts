import { mount } from '@vue/test-utils'
import AtSegmentedControl from '../index'

const factory = (values = {}, slots = { default: [''] }) => {
  return mount(AtSegmentedControl as any, {
    slots,
    props: { ...values },
  })
}

describe('AtSegmentedControl Snap', () => {
  const values = ['tab1', 'tab2', 'tab3']

  it('should render initial AtSegmentedControl', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtSegmentedControl -- props style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtSegmentedControl -- props class', () => {
    const wrapper = factory({ class: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtSegmentedControl -- props values', () => {
    const wrapper = factory({ values: values })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtSegmentedControl -- props current', () => {
    const wrapper = factory({ current: 2, values: values })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtSegmentedControl -- props fontSize', () => {
    const wrapper = factory({ fontSize: '30', values: values })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtSegmentedControl -- props disabled', () => {
    const wrapper = factory({ disabled: true, values: values })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render AtSegmentedControl -- props selectedColor', () => {
    const wrapper = factory({ selectedColor: '#fff', values: values })
    expect(wrapper.element).toMatchSnapshot()
  })
})
