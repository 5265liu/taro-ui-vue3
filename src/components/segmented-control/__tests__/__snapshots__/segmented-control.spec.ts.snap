// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props class 1`] = `
<view
  class="test at-segmented-control test"
  style="border-color: #6190e8;"
/>
`;

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props current 1`] = `
<view
  class="at-segmented-control"
  style="border-color: #6190e8;"
>
  <view
    class="at-segmented-control__item"
    style="color: rgb(97, 144, 232); font-size: 0.5974rem; border-color: #6190e8; background-color: rgb(255, 255, 255);"
  >
    tab1
  </view>
  <view
    class="at-segmented-control__item"
    style="color: rgb(97, 144, 232); font-size: 0.5974rem; border-color: #6190e8; background-color: rgb(255, 255, 255);"
  >
    tab2
  </view>
  <view
    class="at-segmented-control__item at-segmented-control__item--active"
    style="color: rgb(255, 255, 255); font-size: 0.5974rem; border-color: #6190e8; background-color: rgb(97, 144, 232);"
  >
    tab3
  </view>
</view>
`;

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props disabled 1`] = `
<view
  class="at-segmented-control at-segmented-control--disabled"
  style="border-color: #6190e8;"
>
  <view
    class="at-segmented-control__item at-segmented-control__item--active"
    style="color: rgb(255, 255, 255); font-size: 0.5974rem; border-color: #6190e8; background-color: rgb(97, 144, 232);"
  >
    tab1
  </view>
  <view
    class="at-segmented-control__item"
    style="color: rgb(97, 144, 232); font-size: 0.5974rem; border-color: #6190e8; background-color: rgb(255, 255, 255);"
  >
    tab2
  </view>
  <view
    class="at-segmented-control__item"
    style="color: rgb(97, 144, 232); font-size: 0.5974rem; border-color: #6190e8; background-color: rgb(255, 255, 255);"
  >
    tab3
  </view>
</view>
`;

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props fontSize 1`] = `
<view
  class="at-segmented-control"
  style="border-color: #6190e8;"
>
  <view
    class="at-segmented-control__item at-segmented-control__item--active"
    style="color: rgb(255, 255, 255); font-size: 0.64rem; border-color: #6190e8; background-color: rgb(97, 144, 232);"
  >
    tab1
  </view>
  <view
    class="at-segmented-control__item"
    style="color: rgb(97, 144, 232); font-size: 0.64rem; border-color: #6190e8; background-color: rgb(255, 255, 255);"
  >
    tab2
  </view>
  <view
    class="at-segmented-control__item"
    style="color: rgb(97, 144, 232); font-size: 0.64rem; border-color: #6190e8; background-color: rgb(255, 255, 255);"
  >
    tab3
  </view>
</view>
`;

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props selectedColor 1`] = `
<view
  class="at-segmented-control"
  style="border-color: #fff;"
>
  <view
    class="at-segmented-control__item at-segmented-control__item--active"
    style="color: rgb(255, 255, 255); font-size: 0.5974rem; border-color: #fff; background-color: rgb(255, 255, 255);"
  >
    tab1
  </view>
  <view
    class="at-segmented-control__item"
    style="color: rgb(255, 255, 255); font-size: 0.5974rem; border-color: #fff; background-color: rgb(255, 255, 255);"
  >
    tab2
  </view>
  <view
    class="at-segmented-control__item"
    style="color: rgb(255, 255, 255); font-size: 0.5974rem; border-color: #fff; background-color: rgb(255, 255, 255);"
  >
    tab3
  </view>
</view>
`;

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props style 1`] = `
<view
  class="at-segmented-control"
  style="color: red; border-color: #6190e8;"
/>
`;

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props values 1`] = `
<view
  class="at-segmented-control"
  style="border-color: #6190e8;"
>
  <view
    class="at-segmented-control__item at-segmented-control__item--active"
    style="color: rgb(255, 255, 255); font-size: 0.5974rem; border-color: #6190e8; background-color: rgb(97, 144, 232);"
  >
    tab1
  </view>
  <view
    class="at-segmented-control__item"
    style="color: rgb(97, 144, 232); font-size: 0.5974rem; border-color: #6190e8; background-color: rgb(255, 255, 255);"
  >
    tab2
  </view>
  <view
    class="at-segmented-control__item"
    style="color: rgb(97, 144, 232); font-size: 0.5974rem; border-color: #6190e8; background-color: rgb(255, 255, 255);"
  >
    tab3
  </view>
</view>
`;

exports[`AtSegmentedControl Snap render initial AtSegmentedControl 1`] = `
<view
  class="at-segmented-control"
  style="border-color: #6190e8;"
/>
`;
